### PR TITLE
4118 approval

### PR DESCRIPTION
**Pull request by shabeeb-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4118

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/views/auth/staff/ReviewAccountView.vue [hunk 1]
```diff
@@ -108,112 +112,73 @@ import { getModule } from 'vuex-module-decorators'
...
+  public isSaving = false
+
   private readonly pagesEnum = Pages
-  private readonly accountStatusEnum = AccountStatus
+  private readonly TaskRelationshipStatusEnum = TaskRelationshipStatus
 
   private isConfirmationModal:boolean = false
   private isRejectModal:boolean = false
+  public task :Task
+  public taskrRelationshipType:string = ''
```
Comment (karthik-aot) [inline auth-web/src/views/auth/staff/ReviewAccountView.vue:146]: typo in the name :)

File: auth-web/src/services/task.services.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    if (taskFilter.pageLimit) {
+      params.append('limit', taskFilter.pageLimit.toString())
+    }
+
+    return axios.get(`${ConfigHelper.getAuthAPIUrl()}/tasks`, { params })
+  }
+
+  static async approvePendingTask (task:any): Promise<AxiosResponse> {
+    const taskId = task.id
+    return axios.put(`${ConfigHelper.getValue('VUE_APP_AUTH_ROOT_API')}/tasks/${taskId}`, { relationshipStatus: 'ACTIVE' })
```
Comment (karthik-aot) [inline auth-web/src/services/task.services.ts:32]: can we pass the enum

**Review by karthik-aot — CHANGES_REQUESTED** [review]

**Review by saravanpa-aot — APPROVED** [review]

File: auth-api/src/auth_api/services/products.py [hunk 3]
```diff
@@ -93,14 +93,16 @@ def create_product_subscription(org_id, subscription_data: Dict[str, Any],  # py
                 # create a staff review task for this product subscription if pending status
                 if product_model.default_subscription_status == ProductSubscriptionStatus.PENDING_STAFF_REVIEW.value:
                     user = UserModel.find_by_jwt_token(token=token_info)
-                    task_info = {'name': f'{org.name} - {product_model.description}',
+                    task_type = current_app.config.get('ACCESS_REQUEST_PRODUCT')
```
Comment (mengdong19) [inline auth-api/src/auth_api/services/products.py:96]: Need some further thinking about how to save these values/descriptions. enum or config or some sort of database table.

File: auth-api/src/auth_api/config.py [hunk 4]
```diff
@@ -171,6 +171,11 @@ class _Config():  # pylint: disable=too-few-public-methods
     API_GW_EMAIL_SUFFIX = os.getenv('API_GW_EMAIL_SUFFIX', None)
     API_GW_KC_CLIENT_ID_PATTERN = os.getenv('API_GW_KC_CLIENT_ID_PATTERN', 'api-key-account-{account_id}')
 
+    # Type column prefix text in Tasks table - "New Account" for approving affidavit &
+    # "Access request" for Product request
+    NEW_ACCOUNT_STAFF_REVIEW = 'NEW ACCOUNT'
```
Comment (mengdong19) [inline auth-api/src/auth_api/config.py:176]: Should these be hardcoded or put in env config?

**Review by mengdong19 — APPROVED** [review]

File: auth-api/src/auth_api/services/products.py (hunk 3 above)
Comment (karthik-aot) [reply auth-api/src/auth_api/services/products.py]: Refactored them to enums

File: auth-api/src/auth_api/config.py (hunk 4 above)
Comment (karthik-aot) [reply auth-api/src/auth_api/config.py]: Refactored them to enums

**Pull request merged**